### PR TITLE
Ops-847_create_agreement_flow

### PR DESCRIPTION
**Pull request by tdonaworth**
## What changed

- Created initial `/agreements/create` pages and flow
- Created `POST` for `/agreements` endpoint
- updated unit tests
- many bug fixes
- many format fixes

## Issue

#847 
#833 
#835 
#859 
...and many more

## How to test

- Unit Tests for now, or visit `/agreements/create` in the frontend

### CODE REVIEW

**Comment by jonnalley** [comment]
I'm assuming the this PR covers all work intended for the above-referenced tasks such that no further work needed outside of this PR and will mark them as done/closed accordingly.

**Comment by fpigeonjr** [comment]
workin on the conflicts now

**Review by fpigeonjr — APPROVED** [review]
works well in`localdev`

File: frontend/src/components/ResearchProjects/ProjectSelect/ProjectSelect.jsx [hunk 1]
```diff
@@ -0,0 +1,141 @@
+import { useDispatch, useSelector } from "react-redux";
+
+export const ProjectSelect = ({ researchProjectsState, selectedProjectState, setSelectedProjectFunction }) => {
+    const dispatch = useDispatch();
+    const researchProjects = useSelector((state) => state[researchProjectsState]);
+    const selectedResearchProject = useSelector((state) => state[selectedProjectState]);
+
+    console.log(`Project-List: ${researchProjects}`);
```
Comment (johndeange) [inline frontend/src/components/ResearchProjects/ProjectSelect/ProjectSelect.jsx:8]: Would be good to remove these log statements.

File: frontend/src/components/UI/Header/Menu.jsx [hunk 2]
```diff
@@ -23,6 +23,9 @@ export const Menu = () => {
                 <li className="usa-nav__primary-item">
                     <Link to="/budget-lines/create">Create Budget Line</Link>
                 </li>
+                <li className="usa-nav__primary-item">
+                    <Link to="/agreements/create">Create Agreement</Link>
```
Comment (johndeange) [inline frontend/src/components/UI/Header/Menu.jsx:27]: I wonder what the Breadcrumb component should be for these create features?

File: frontend/src/pages/agreements/CreateAgreement.jsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import { getProcurementShopList } from "../../api/getProcurementShopList";
+import { StepSelectProject } from "./StepSelectProject";
+import { StepCreateAgreement } from "./StepCreateAgreement";
+import { StepCreateBudgetLines } from "./StepCreateBudgetLines";
+
+const wizardSteps = ["Project", "Agreement", "Budget Lines"];
+
+export const CreateAgreement = () => {
+    const dispatch = useDispatch();
+    //const selectedProject = useSelector((state) => state.createBudgetLine.selectedProject);
```
Comment (johndeange) [inline frontend/src/pages/agreements/CreateAgreement.jsx:15]: remove comments

File: frontend/src/pages/agreements/StepCreateAgreement.jsx [hunk 4]
```diff
@@ -0,0 +1,157 @@
+import { useSelector, useDispatch } from "react-redux";
+import { StepIndicator } from "../../components/UI/StepIndicator/StepIndicator";
+import { ProcurementShopSelect } from "./ProcurementShopSelect";
+import { AgreementReasonSelect } from "./AgreementReasonSelect";
+import { AgreementTypeSelect } from "./AgreementTypeSelect";
+import { ProductServiceCodeSelect } from "./ProductServiceCodeSelect";
+import {
+    setProcurementShopsList,
+    setSelectedProcurementShop,
```
Comment (johndeange) [inline frontend/src/pages/agreements/StepCreateAgreement.jsx:9]: Address linting issues

**Review by johndeange — APPROVED** [review]

**Pull request merged**